docs/api.rst

```diff
@@ -536,10 +536,59 @@ Transport exceptions
 - ``MessageReceiveError``: The transport encountered any other error while trying to receive a message.
 
 
-ASGI Transport
-++++++++++++++
+Redis Gateway Transport
++++++++++++++++++++++++
+
+The ``transport.redis_gateway`` module provides a transport implementation that uses Redis (in simple or Sentinel mode)
+for sending and receiving messages. This is the recommended transport for use with ``pysoa``, as it provides a
+convenient and performant backend for asynchronous service requests.
+
+Standard and Sentinel modes
+***************************
+
+The Redis Gateway transport has two primary modes of operation: in "standard" mode, the channel layer will connect to a
+specified list of Redis hosts, while in "Sentinel" mode, the channel layer will connect to a list of Sentinel hosts and
+use Sentinel to find its Redis hosts.
+
+Configuration
+*************
+
+The Redis Gateway transport takes the following extra keyword arguments for configuration:
+
+- ``backend_type``: Either "redis.standard" or "redis.sentinel" to specify which Redis backend to use (required)
+
+- ``backend_layer_kwargs``: A dictionary of arguments to pass to the backend layer
+
+  + ``connection_kwargs``: A dictionary of arguments to pass to the underlying Redis client
+
+  + ``hosts``: A list of strings (host names / IP addresses) or tuples (host names / IP addresses and ports) for Redis hosts or sentinels to which to connect (will use "localhost" by default)
+
+  + ``redis_db``: The Redis database number to use (a shortcut for specifying ``connection_kwargs['db']``)
+
+  + ``redis_port``: The connection port to use (a shortcut for providing this for every entry in ``hosts``
+
+  + ``sentinel_refresh_interval``: How often, in seconds, master/slave data should be refreshed from the Sentinel (only for type "redis.sentinel") (refreshes every connection by default)
```

Comment:
You use a persistent connection instead of reopening a connection for every operation, right? Does this mean the default is to refresh every time a connection is opened or every time a connection is used?

Do you refresh the connections when a redis or network related exception occurs? That's what is most important, not periodically refreshing, imho.

Reply:
Yes, a persistent connection is used, from a pool. We get a connection from the pool on every Redis call. For sending to a server or receiving from a client, this is just round-robin whichever connection we get next. For sending to a client or receiving from a server, this is consistently hashed (always the same connection for a given client ID). The default is to refresh every time we get a connection from the pool, not every time we open a new connection.

We do not refresh the connections at this layer when a Redis- or network-related exception occurs. On the client side, that request fails and the client is re-created for the next request, which results in the same effect as refreshing connections. On the server side, I'm not sure how we handle it. But I would note that this does not reflect a change in behavior from the current behavior (and, in this PR, I'm just documenting this; the new transport was added in an earlier PR).